pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/timeseries/PhysicalTimeSeriesServerPlanVisitor.java

```diff
@@ -42,17 +42,14 @@
 import org.apache.pinot.tsdb.spi.plan.LeafTimeSeriesPlanNode;
 
 
-public class PhysicalTimeSeriesPlanVisitor {
-  public static final PhysicalTimeSeriesPlanVisitor INSTANCE = new PhysicalTimeSeriesPlanVisitor();
-
+public class PhysicalTimeSeriesServerPlanVisitor {
   private QueryExecutor _queryExecutor;
   private ExecutorService _executorService;
   private ServerMetrics _serverMetrics;
 
-  private PhysicalTimeSeriesPlanVisitor() {
-  }
-
-  public void init(QueryExecutor queryExecutor, ExecutorService executorService, ServerMetrics serverMetrics) {
+  // Warning: Don't use singleton access pattern, since Quickstarts run in a single JVM and spawn multiple broker/server
```

Comment:
Is this issue visible when multiple broker instances are running?

Reply:
The issue only manifests in Quickstarts when numServers > 1. The previous pattern of using an INSTANCE variable leads to hard to debug errors, so left a Warning for future readers.